compiler/optimizer/VPConstraint.hpp

```diff
@@ -310,6 +331,34 @@ class VPConstraint
       return diff;
       }
 
+   float floatAddWithOverflow(float a, float b, bool& overflow)
+      {
+      float sum = TR::Compiler->arith.floatAddFloat(a, b);
+      overflow = (sum == FLOAT_POS_INFINITY) || (sum == FLOAT_NEG_INFINITY);
```

Comment:
Could you please elaborate on your reasoning behind considering this an adequate condition for detecting overflow in these new routines you added in this commit ?

Reply:
According to IEEE 754, whenever there is an overflow the result returned is INFINITY